src/py/flwr/server/superlink/state/in_memory_state.py

```diff
@@ -189,6 +189,15 @@ def get_task_res(self, task_ids: set[UUID], limit: Optional[int]) -> list[TaskRe
             replied_task_ids: set[UUID] = set()
             for _, task_res in self.task_res_store.items():
                 reply_to = UUID(task_res.task.ancestry[0])
+                task_ins = self.task_ins_store.get(reply_to)
+                if task_ins is None:
+                    log(ERROR, "TaskIns with task_id %s does not exist.", reply_to)
+                    return []
```

Comment:
We should not return `[]` here. Some of the requested `task_ids` might have responses and we should return those back to the caller. In the future, we need to introduce a way to return errors for those IDs that are either not existing or expired.